### PR TITLE
Added a feature to be able to select the appliance to connect by action

### PR DESCRIPTION
**Pull request by userlocalhost**
### Background
This pack has the following subjects.
* StackStorm can't load config.schema file because of the wrong suffix of it.
* The target appliance which is handled by each actions is static (as reported by #4).

### Action
These are main points of this patch.
* Renaming the file extension of config.schema file from 'yml' to 'yaml'.
* Adding a parameter 'appliance' to specify the appliance to connect for each actions.

### CODE REVIEW

File: actions/add_slb_server.yml [hunk 1]
```diff
@@ -37,3 +37,6 @@ parameters:
           - True
           - False
         default: False
+    appliance:
+        required: true
+        default: default
```
Comment (LindsayHill) [inline actions/add_slb_server.yml]: I think that this parameter should declare a type, for example: `type: string`.

I also think that it should not be required. I don't think it makes sense to have it `required: true` AND have a default value.

(Same comments apply for all the places you've added the `appliance` parameter

File: config.schema.yml [hunk 2]
```diff
@@ -1,17 +0,0 @@
----
```
Comment (LindsayHill) [inline config.schema.yml:1]: I don't know how I missed this the first time around. Sorry, I should have identified it when reviewing. 

There was some inconsistency in the past with what ST2 allowed, for `yml` vs `yaml`. But it should be all `.yaml` now.

File: actions/add_slb_service_group.yml [hunk 3]
```diff
@@ -57,3 +57,6 @@ parameters:
             - dst-ip-only-hash
             - dst-ip-hash
         default: round-robin
+    appliance:
+        required: true
+        default: default
```
Comment (LindsayHill) [inline actions/add_slb_service_group.yml]: I think that all of these action metadata files should be `.yaml`, not `.yml`

File: config.schema.yaml [hunk 4]
```diff
@@ -0,0 +1,20 @@
+---
+appliance:
+  type: object
+  properties:
+    default:
```
Comment (LindsayHill) [inline config.schema.yaml]: I don't _think_ this is right. I think it works if you have just one `default` appliance setting, but it won't work if you declare multiple appliance settings. I think it needs to look more like this: https://github.com/StackStorm-Exchange/stackstorm-mysql/blob/master/config.schema.yaml

File: config.schema.yml (hunk 2 above)
Comment (userlocalhost) [reply config.schema.yml:1]: > But it should be all .yaml now.

I got it. I'll change the suffix of all actions and configurations to `yaml`.

**Comment by userlocalhost** [comment]
Thank you very much for your review and giving some important advices.
I'll fix them soon.

File: config.schema.yaml (hunk 4 above)
Comment (userlocalhost) [reply config.schema.yaml]: Thank you for your advice @LindsayHill . I corrected it in [this commit](https://github.com/StackStorm-Exchange/stackstorm-acos/pull/6/commits/bbe76c98e556573ad3a2f01f9ab09e76c88685c7). But I'm not sure that is the best one.

I understood that I should have to write the config.schema to be able to specify multiple configuration-sets (`target IP address`, `username` and `password`) to connect. And I corrected `config.schema.yaml` following the one of MySQL pack.

But I have no confidence that the corrected `config.schema.yaml` is right.

In the `st2 config mysql` command, the properties which are specified in the `config.schema.file` (e.g. `host`, `user`, ...) aren't asked interactively. This is counterintuitive, I guess.

I'm not familiar with the implementation to parse the configuration schema file. So I'll make sure to know it.
If you know the best way to do it, could you please tell me that?

**Comment by userlocalhost** [comment]
In the last commit, I corrected this patch as following.

* Replaced the specification of API version parameter from action metadata to the appliance setting in the configuration file because it is related to the appliance.
* Allowed to set multiple setting of appliance information in the configuration file and User can specify it from action parameter.

**Comment by userlocalhost** [comment]
I fixed remaining works below which were associated with the change of this PR.
- Removing old-style default configuration value (6411aa8)
- Filtering extended parameters that are not used by ACOS Client (cb8a249)

**Review by LindsayHill — APPROVED** [review]
Looking good - ready to merge now?

**Comment by userlocalhost** [comment]
Thank you for your review and fixing. Yes, it is!

**Pull request merged**